packages/cart/src/types/line-item.ts

```diff
@@ -0,0 +1,35 @@
+interface PartialUpsertLineItemDTO {
+  subtitle?: string
+  thumbnail?: string
+
+  product_id?: string
+  product_title?: string
+  product_description?: string
+  product_subtitle?: string
+  product_type?: string
+  product_collection?: string
+  product_handle?: string
+
+  variant_id?: string
+  variant_sku?: string
+  variant_barcode?: string
+  variant_title?: string
+  variant_option_values?: Record<string, unknown>
+
+  requires_shipping?: boolean
+  is_discountable?: boolean
+  is_tax_inclusive?: boolean
+
+  compare_at_unit_price?: number
+}
+
+export interface CreateLineItemDTO extends PartialUpsertLineItemDTO {
+  title: string
+  quantity: number
+  unit_price: number
+  cart_id: string
+}
+
+export interface UpdateLineItemDTO extends PartialUpsertLineItemDTO {
```

Comment:
question: aren't we missing the `extends PartialUpsertLineItemDTO, Partial<CreateLineItemDTO>` ?

Reply:
Nice catch!